src/components/app-content-renderer/recommendation-tile.js

```diff
@@ -21,30 +13,28 @@ import {
 import { useIntl } from 'react-intl';
 
 import useRequest from './use-request';
-
-const NoIcon = () => <span className="ins-c-no-icon">No icon</span>;
-
-const groupIconMapper = {
-  automation: ProcessAutomationIcon,
-  exclamationTriangle: ExclamationTriangleIcon,
-  exclamationCircle: ExclamationCircleIcon,
-  infoCircle: InfoCircleIcon,
-  checkCircle: CheckCircleIcon,
-  security: SecurityIcon,
-  unknown: QuestionCircleIcon,
-  default: NoIcon,
-};
+import iconMapper from '../../utils/icon-mapper';
+import { useDispatch } from 'react-redux';
+import { removeRecommendationTile } from '../../store/actions';
 
 const RecommendationGroup = (recommendation) => {
   const intl = useIntl();
+  const dispatch = useDispatch();
   const [{ count, response, show }] = useRequest(recommendation);
 
+  useEffect(() => {
+    if (show === false) {
+      dispatch(
+        removeRecommendationTile(recommendation.id, recommendation.recId)
```

Comment:
You can use `onRequest` argument in the hook to do this. See `FirstPanelTile`.

Reply:
Yeah, we will also need the on error callback with the same code. I've realized it yesterday evening.

Reply:
Yeah, that's the third argument `onError` for! 😎